src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -860,7 +860,8 @@ private void GetDisposalInfoForEnumerator(ref ForEachEnumeratorInfo.Builder buil
             {
                 builder.NeedsDisposal = true;
             }
-            else if (enumeratorType.IsRefLikeType || isAsync)
+            else if (Compilation.IsFeatureEnabled(MessageID.IDS_FeatureUsingDeclarations) &&
```

Comment:
>Compilation.IsFeatureEnabled [](start = 21, length = 28)

Is there a test for this case?

Reply:
TestPatternDisposeRefStruct / TestPatternDisposeRefStruct_7_3  [http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests/Semantics/ForEachTests.cs,3173](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests/Semantics/ForEachTests.cs,3173)

---
In reply to: [349348704](https://github.com/dotnet/roslyn/pull/39949#discussion_r349348704) [](ancestors = 349348704)